src/zope/app/wsgi/testlayer.py

```diff
@@ -241,6 +244,7 @@ def request(self, host, handler, request_body, verbose=0):
                 dict(extra_headers)["Authorization"],)
 
         request += "\n" + request_body
+        # XXX: http() needs to be passed a wsgi app!  where do we get a wsgi app?
```

Comment:
This comment confuses me.

Reply:
The signature of http() is

```
def http(wsgi_app, string, handle_errors=True):
```

The line below my XXX comment is

```
        response = http(request, handle_errors=self.handleErrors)
```

You can see how that cannot possibly work.

I think this situation is due to historical reasons: we used to have a `http` in zope.app.testing.functional that was an instance of HTTPCaller and had a `__call__` that took `request` and `handle_errors`.  The new `http` defined here requires an extra argument, `wsgi_app`, but nobody noticed that `XMLRPCTestTransport.request` is now broken.

(I don't know how to fix it.  zope.testbrowser.testing.Layer.get_app() is a class method that returns the current WSGI app in a test that uses a zope.testbrowser test layer for functional tests.  It doesn't seem appropriate to make zope.app.wsgi depend on zope.testbrowser in this way.)

Reply:
Ah, okay, so you want to tell that the call is broken. Is there a GitHub issue which could be referenced in the comment?

Additionally I do not think that using `zope.testbrowser.testing.Layer.get_app()` is a good idea as it couples the code way too tight to this one implementation in `zope.testbrowser`.

Reply:
Yeah, I ought to open an issue for that.

(Unless somebody else preempts me ;)